From e1b4d9fd5b796711b38475c381a168a99003163c Mon Sep 17 00:00:00 2001
From: Juergen Repp <juergen.repp@sit.fraunhofer.de>
Date: Thu, 2 Dec 2021 09:17:15 +0100
Subject: [PATCH 09/23] FAPI: Fix leak in fapi crypto with ossl3

A leak in the case "out of memory" detected by scan-build was fixed.

Signed-off-by: Juergen Repp <juergen.repp@sit.fraunhofer.de>
---
 src/tss2-fapi/fapi_crypto.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tss2-fapi/fapi_crypto.c b/src/tss2-fapi/fapi_crypto.c
index d061cf48..fd7ea555 100644
--- a/src/tss2-fapi/fapi_crypto.c
+++ b/src/tss2-fapi/fapi_crypto.c
@@ -1573,7 +1573,7 @@ ifapi_crypto_hash_start(IFAPI_CRYPTO_CONTEXT_BLOB **context,
      * As we don't want the TPM to be called for these operations, we have
      * to initialize own library context with the default provider. */
     mycontext->libctx = OSSL_LIB_CTX_new();
-    return_if_null(mycontext->libctx, "Out of memory", TSS2_FAPI_RC_MEMORY);
+    goto_if_null(mycontext->libctx, "Out of memory", TSS2_FAPI_RC_MEMORY, cleanup);
 
     if (!(mycontext->osslHashAlgorithm =
             EVP_MD_fetch(mycontext->libctx, get_hash_md(hashAlgorithm), NULL))) {
-- 
2.34.3

